src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/UploadDocumentsHandler.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    private static final String SUPPLIED_BY_APPLICANT = "applicant";
+    private static final String SUPPLIED_BY_LEGAL_REPRESENTATIVE = "legalRepresentative";
+    private static final String SUPPLIED_BY_HOME_OFFICE = "homeOffice";
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<BailCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.UPLOAD_DOCUMENTS;
+    }
+
+    public PreSubmitCallbackResponse<BailCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<BailCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final BailCase bailCase = callback.getCaseDetails().getCaseData();
+        String userRole = bailCase.read(CURRENT_USER, String.class).orElse("");
+
+        Optional<List<IdValue<DocumentWithDescription>>> maybeDocument = bailCase.read(UPLOAD_DOCUMENTS);
+        Optional<String> maybeSuppliedBy = bailCase.read(UPLOAD_DOCUMENTS_SUPPLIED_BY);
+
+        String suppliedBy = maybeSuppliedBy
+            .orElse(currentUserIsLegalRep(userRole)
+                        ? "Legal Representative"
+                        : currentUserIsHomeOfficer(userRole)
+                ? "Home Office"
+                : "");
+
+        if (maybeDocument.isPresent()) {
+            List<DocumentWithMetadata> document =
+                maybeDocument
+                    .orElseThrow(() -> new IllegalStateException("document is not present"))
+                    .stream()
+                    .map(IdValue::getValue)
+                    .map(doc -> documentReceiver.tryReceive(doc, DocumentTag.UPLOAD_DOCUMENT, suppliedBy))
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collect(Collectors.toList());
+
+            BailCaseFieldDefinition appropriateCollectionForUserRole =
+                selectAppropriateDocumentsCollection(userRole, suppliedBy);
+
+            Optional<List<IdValue<DocumentWithMetadata>>> maybeExistingDocumentsCollection =
+                bailCase.read(appropriateCollectionForUserRole);
+
+            final List<IdValue<DocumentWithMetadata>> existingDocuments =
+                maybeExistingDocumentsCollection.orElse(Collections.emptyList());
+
+            List<IdValue<DocumentWithMetadata>> allDocuments =
+                documentsAppender.append(existingDocuments, document);
+
+            bailCase.write(appropriateCollectionForUserRole, allDocuments);
+
+            bailCase.clear(UPLOAD_DOCUMENTS);
+            bailCase.clear(UPLOAD_DOCUMENTS_SUPPLIED_BY);
+            bailCase.clear(CURRENT_USER);
```

Comment:
Will the screens still work if you clear this field here? Better clear in the confirmation Handler?